android/src/main/java/com/swmansion/gesturehandler/core/GestureHandler.kt

```diff
@@ -192,10 +188,11 @@ open class GestureHandler<ConcreteGestureHandlerT : GestureHandler<ConcreteGestu
 
   protected open fun onPrepare() {}
 
-  private fun getWindow(context: Context?): Window? {
+  private fun getActivity(context: Context?): Activity? {
     if (context == null) return null
-    if (context is Activity) return context.window
-    if (context is ContextWrapper) return getWindow(context.baseContext)
+    if (context is ReactContext) return context.currentActivity
+    if (context is Activity) return context
+    if (context is ContextWrapper) return getActivity(context.baseContext)
```

Comment:
```suggestion
    return when(context){
      is ReactContext -> context.currentActivity
      is Activity -> context
      is ContextWrapper -> getActivity(context.baseContext)
      else -> null
    }
```

Wouldn't this work? I think it looks cleaner.

Reply:
Done in https://github.com/software-mansion/react-native-gesture-handler/pull/2826/commits/422aa713724fc9dfad11757af2259c5cad7b2989